src/main/scala/firrtl/ir/StructuralHash.scala

```diff
@@ -115,7 +115,9 @@ object StructuralHash {
 }
 
 trait HashCode {
-  protected val str: String
+
+  /** string representation of the hash code */
+  def str: String
```

Comment:
Why not `override def toString` instead of calling it `str`?

Reply:
Because this is a _special_ String :)
It essentially satisfies the property that if (and only if) the two hash codes are the same, the two `str` are the same. This isn't generally true for `toString`.

Do you want me to rename it to `toHashString` or something like that?